src/Symfony/Component/HttpClient/Internal/CurlClientState.php

```diff
@@ -52,8 +52,8 @@ public function __construct(int $maxHostConnections, int $maxPendingPushes)
         if (\defined('CURLMOPT_MAX_HOST_CONNECTIONS')) {
             $maxHostConnections = curl_multi_setopt($this->handle, \CURLMOPT_MAX_HOST_CONNECTIONS, 0 < $maxHostConnections ? $maxHostConnections : \PHP_INT_MAX) ? 0 : $maxHostConnections;
```

Comment:
I think `0` is the issue. It's not documented as meaning "infinite", unlike many other curl options.

```suggestion
            $maxHostConnections = curl_multi_setopt($this->handle, \CURLMOPT_MAX_HOST_CONNECTIONS, 0 < $maxHostConnections ? $maxHostConnections : \PHP_INT_MAX) ? \PHP_INT_MAX : $maxHostConnections;
```

Reply:
@nicolas-grekas Yes, you're right, see #58278. I opened it as a bug fix, if it's not, please let me know.